src/nplinker/genomics/bigscape/runbigscape.py

```diff
@@ -15,22 +15,65 @@ def run_bigscape(
     antismash_path: str | PathLike,
     output_path: str | PathLike,
     extra_params: str,
-):
-    bigscape_py_path = "bigscape.py"
-    logger.info(
-        f'run_bigscape: input="{antismash_path}", output="{output_path}", extra_params={extra_params}"'
-    )
+    version: int = 1,
+) -> bool:
+    """Runs BiG-SCAPE to cluster BGCs.
+
+    The behavior of this function is slightly different depending on the version of
+    BiG-SCAPE that is set to run using the configuration file.
+    Mostly this means a different set of parameters is used between the two versions.
+
+    The AntiSMASH output directory should be a directory that contains GBK files.
+    The directory can contain subdirectories, in which case BiG-SCAPE will search
+    recursively for GBK files.
+
+    By default, only GBK Files with "cluster" or "region" in the filename are
+    accepted. GBK Files with "final" in the filename are excluded.
+
+    Args:
+        antismash_path: Path to the antismash output directory.
+        output_path: Path to the output directory where BiG-SCAPE will write its results.
+        extra_params: Additional parameters to pass to BiG-SCAPE.
+        version: The version of BiG-SCAPE to run. Can be 1 or 2.
+
+    Returns:
+        True if BiG-SCAPE ran successfully, False otherwise.
+    """
+    # switch to correct version of BiG-SCAPE
+    if version == 1:
+        bigscape_py_path = "bigscape.py"
+    elif version == 2:
+        bigscape_py_path = "bigscape-v2.py"
+    else:
+        raise ValueError("Unexpected BiG-SCAPE version number specified")
 
     try:
         subprocess.run([bigscape_py_path, "-h"], capture_output=True, check=True)
     except Exception as e:
-        raise Exception(f"Failed to find/run bigscape.py (path={bigscape_py_path}, err={e})") from e
+        raise FileNotFoundError(
+            f"Failed to find/run bigscape.py (path={bigscape_py_path}, err={e})"
+        ) from e
 
     if not os.path.exists(antismash_path):
-        raise Exception(f'antismash_path "{antismash_path}" does not exist!')
+        raise FileNotFoundError(f'antismash_path "{antismash_path}" does not exist!')
 
-    # configure the IO-related parameters, including pfam_dir
-    args = [bigscape_py_path, "-i", antismash_path, "-o", output_path, "--pfam_dir", PFAM_PATH]
+    logger.info(f"Running BiG-SCAPE version {version}")
+    logger.info(
+        f'run_bigscape: input="{antismash_path}", output="{output_path}", extra_params={extra_params}"'
+    )
+
+    # assemble arguments. first argument is the python file
+    args = [bigscape_py_path]
+
+    # version 2 points to specific Pfam file, version 1 points to directory
+    # version 2 also requires the cluster subcommand
+    if version == 1:
+        args.extend(["--pfam_dir", PFAM_PATH])
+    elif version == 2:
+        args.extend(["cluster", "--pfam_path", os.path.join(PFAM_PATH, "Pfam-A.hmm")])
```

Comment:
Since some bigscape parameters have been pre-configured, it's needed to update `data/nplinker.toml` to inform users not to set these parameters.

Reply:
If you mean the "cluster" bit, that's the mode in which BiG-SCAPE v2 runs and separates some functionality that were previously extra parameters.
In particular, query mode is no longer a parameter, but it's own mode run using `bigscape.py query [params]`.

I will add this to the explanation of nplinker.toml

Reply:
There are explanations for version 1 in [nplinker.toml](https://github.com/NPLinker/nplinker/blob/1e23e57a81910ad19964d4c65b0d67be12550362/src/nplinker/data/nplinker.toml#L44), you could take a look. 
If version 2 has subcommands, then we need to make it clear it's using the subcommand `cluster` and the parameter `--pfam_path`.